src/vision.cs

```diff
@@ -672,19 +741,30 @@ interface VNDetectBarcodesRequest {
 		[DesignatedInitializer]
 		IntPtr Constructor ([NullAllowed] VNRequestCompletionHandler completionHandler);
 
+		[Deprecated (PlatformName.MacOSX, 12, 0, message: "Use 'GetSupportedSymbologiesAndReturnError' instead.")]
+		[Deprecated (PlatformName.iOS, 15, 0, message: "Use 'GetSupportedSymbologiesAndReturnError' instead.")]
+		[Deprecated (PlatformName.TvOS, 15, 0, message: "Use 'GetSupportedSymbologiesAndReturnError' instead.")]
 		[Static]
 		[Protected]
 		[Export ("supportedSymbologies", ArgumentSemantic.Copy)]
 		NSString [] WeakSupportedSymbologies { get; }
 
-		[Static]
 		[Wrap ("VNBarcodeSymbologyExtensions.GetValues (WeakSupportedSymbologies)")]
 		VNBarcodeSymbology [] SupportedSymbologies { get; }
 
+		[TV (15,0), Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+		[Export ("supportedSymbologiesAndReturnError:")]
+		[return: NullAllowed]
+		string[] GetSupportedSymbologiesAndReturnError ([NullAllowed] out NSError error);
```

Comment:
remove `AndReturnError` (and fix the earlier deprecation to match)

Reply:
return `VNBarcodeSymbology[]`

Reply:
I keep running into this error when I add `[return: BindAs (typeof (VNBarcodeSymbology []))]` and/or switch it to `VNBarcodeSymbology[] GetSupportedSymbologiesAndReturnError`. Do you know what might be the issue here?
```
build/tvos/tvos/Vision/VNDetectBarcodesRequest.g.cs(91,18): error CS0452: The type 'VNBarcodeSymbology' must be a reference type in order to use it as parameter 'T' in the generic type or method 'CFArray.ArrayFromHandle<T>(IntPtr)'
```

Reply:
that's not a very common scenario so it might not be supported by the generator
you'll need to bind (in can be `[Internal]`) a version that returns `NSString[]`
then you can add a version that converts (each of) them into the enum array: `VNBarcodeSymbology []`